en/docs/guides/access-delegation/get-user-claims-as-a-jwt.md

```diff
@@ -0,0 +1,179 @@
+# Get User Claims as a JWT
+
+This page provides instructions on how to get user claims of the authorized user as a [JWT token](insertlink) with the validation response. 
+
+You can configure this for all tenants using the file configurations by configuring the `deployment.toml` file or you can configure it tenant-wise using the management console. 
+
+{!fragments/register-a-service-provider.md!}
+
+3.  Expand **Inbound Authentication Configuration** section and then **OAuth/OpenID Connect Configuration**. 
+
+4. Click **Configure.**   
+ 
+5. Enter the **Callback Url**.
+
+    !!! tip
+        For more information on `Callback Url` field and other advanced configurations, see [Advanced OpenID Connect Configurations](../../authentication/oauth-app-config-advanced).
+
+6. Select **JWT** as the **Token Issuer**.
+ 
+    ![jwt-token-issuer](../../assets/img/guides/jwt-token-issuer.png)
+
+7.  Click **Add**. 
+
+    Note that the **OAuth Client Key** and **Client Secret** are generated. You will need these values later on when sending the request to the authorization endpoint.
+
+8.  Click **Register**.
+
+-----
+
+
+## Using file configurations
+
+**Alternatively**, you can enable user claims to be included in the jwt token across all tenants by configuring the `deployment.toml` file. 
+
+**Mandatory configurations**
+
+1. Add and configure the following properties as shown below in the `deployment.toml` file found in the `<IS_HOME>/repository/conf` folder.
+
+    ```toml
+    [oauth.token.validation]
+    include_validation_context_as_jwt_in_reponse = "true"
+    validation_response_signing_algorithm = "SHA256withRSA"
+    validation_response_jwt_validity = "15m"
+    ```
+
+    ```toml
+    [oauth.extensions]
+    token_context_generator = "org.wso2.carbon.identity.oauth2.authcontext.JWTTokenGenerator"
+    token_context_claim_retriever = "org.wso2.carbon.identity.oauth2.authcontext.DefaultClaimsRetriever"
+    token_context_dialect_uri = "http://wso2.org/claims"
+    ```
+
+2. Add the following property to the `deployment.toml` file to use the JWT Token Builder instead of the default Token Builder.
+
+    ```toml
+    [oauth.token_generation]
+    access_token_type = "self_contained"
+    ```
+
+**Optional configurations**
+
+
+1.  See the [Extension Points for OAuth](insertlink) topic for more details about the usage of the `TokenGeneratorImplClass` and `ClaimsRetrieverImplClass`.
```

Comment:
See [Extension Points for OAuth](insertlink) for more details about the usage of `TokenGeneratorImplClass` and `ClaimsRetrieverImplClass`.